cql3/expr/term_expr.cc

```diff
@@ -294,40 +307,48 @@ set_prepare_term(const collection_constructor& c, database& db, const sstring& k
         // away to simplify predicate evaluation.  See also
         // https://issues.apache.org/jira/browse/CASSANDRA-5141
         if (receiver->type->is_multi_cell()) {
-            return cql3::constants::NULL_VALUE;
+            return constant::make_null(receiver->type);
         }
         // We've parsed empty maps as a set literal to break the ambiguity so
         // handle that case now. This branch works for frozen sets/maps only.
         const map_type_impl* maybe_map_type = dynamic_cast<const map_type_impl*>(receiver->type.get());
         if (maybe_map_type != nullptr) {
-            serialized_compare comparator = maybe_map_type->get_keys_type()->as_less_comparator();
-            std::map<managed_bytes, managed_bytes, serialized_compare> m(empty_type->as_less_comparator());
-            return ::make_shared<maps::value>(std::move(m), receiver->type);
+            collection_constructor map_value {
+                .style = collection_constructor::style_type::map,
+                .elements = {},
+                .type = receiver->type
+            };
+            return expr::evaluate(map_value, query_options::DEFAULT);
```

Comment:
You could have returned an expr::constant here (but this is okay too).

Reply:
I would have to serialize `int32_t(0)` manually and didn't want to do that.
Also this is more future proof - it will work even if serialization format changes.